Add doc for setting up Grafana (#897)
PelicanPlatform/pelican: docs/pages/grafana.mdx

## Configure authentication

This section discribes how to configure Grafana to have proper authentication credentials to access Prometheus instance on Pelican or how to turn off Pelican authentication for Prometheus. Depending on your deployment, you may prefer one way over the other.

### Authenticated Prometheus

By default, Pelican protects the Prometheus endpoint by asking for a [JWT](https://jwt.io/introduction) to verify the user accessing the endpoint. It reads `Bearer <token>` from `Authorization` header for any HTTPS requests to `/api/v1.0/prometheus` and its subpath. You need to create a JWT using Pelican CLI and pass it to Grafana.

1. Run the following command

    * Replace `<server-web-url>` to the URL to your Pelican server Web UI. By default, it's `https://localhost:8444`.

    * Replace `--lifetime 604800` to your desired lifetime for the token. The value is in seconds. The example below generates a token with a lifetime of 7 days.

    ```bash filename="bash"
    $ pelican origin token create --scope monitoring.query --claim wlcg.ver=1.0 --issuer <server-web-url> --subject grafana --audience <server-web-url> --lifetime 604800

>     eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL2V4YW1wbGUuY29tIiwiaWF0IjoxNzA5NjUwMjMyLCJleHAiOjE3NDExODYyMzIsImF1ZCI6Imh0dHBzOi8vZXhhbXBsZS5jb20iLCJzdWIiOiJncmFmYW5hIiwic2NvcGUiOiJtb25pdG9yaW5nLnF1ZXJ5Iiwid2xjZy52ZXIiOiIxLjAifQ.cK5QyYJinn3QCPa6YB10Wn37MEEKzt_kyDMR9meN1Ec

Author: This is a fake token generated from a public web app

The environment variable for the configuration is `PELICAN_MONITORING_PROMQLAUTHORIZATION`.

You can either `export` this environment variable or append it before your command to run Pelican.

```bash
export PELICAN_MONITORING_PROMQLAUTHORIZATION=false
```

or

```bash
PELICAN_MONITORING_PROMQLAUTHORIZATION=false pelican origin serve
```

Once you configured the authentication either in Grafana or in Pelican, click **Save & Test** to see if the connection can be established.

# Build Monitroing Dashbord

With Grafana connected to Pelican Prometheus instance, you are able to build visualization with the metrics Pelican collects. This section describes how to get started with a template dashboard configuration for a _Pelican origin server_. For a detailed guide on how to create a Grafana dashboard, refer to [Grafana documentation](https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/create-dashboard/).

## Before start

1. Make sure your Pelican origin server is running

2. Download the template dashboard file from [here](https://github.com/PelicanPlatform/pelican/blob/main/docs/public/origin-dashboard-template.json)

## Modify dashboard template

> There are a couple of placeholders need to be changed before you can import the dashboard template to Gafana. To do a batch replace, it is recommended that you use a text editor to find and replace placeholders to corresponding values.

Reviewer: ```suggestion
There are a couple of placeholders that need to be changed before you can import the dashboard template to Grafana. To do a batch replace it is recommended that you use a text editor to find and replace placeholders to corresponding values.
```

* Replace `<YOUR-DATA-SOURCE-UID>` to the id of your Prometheus data source. You can find the ID on the data source configuration page.

    1. Navigate to the configuration page
    2. Copy the ID from your browser's URL box, which should have a value similar to

        ```console
        http://localhost:3000/connections/datasources/edit/fc2115e0-ede3-4fdb-9056-ecefa1b3160a

>         ```

Reviewer: Unsure if this is just in my view, but it looks like (at least in Github) that the ``` didn't close the console block here. But again, that could just be GitHub. Everything below it is still the blue of the console block, oddly enough.

Author: That should be fine, my local markdown preview works

        Where `fc2115e0-ede3-4fdb-9056-ecefa1b3160a` is the ID you want to copy. Note that the ID in the example above will be different from your data source ID

>     3. Batch replace `<YOUR-DATA-SOURCE-UID>` in the template dashboard file to the copied value

Reviewer: ```suggestion
    3. Batch replace `<YOUR-DATA-SOURCE-UID>` in the template dashboard file with the copied value
```

> * Batch replace `<YOUR-ORIGIN-SERVER-WEB-URL>` to the web URL of your Pelican origin server, without the `https://` portion. Example: `localhost:8444`

Reviewer: ```suggestion
* Batch replace `<YOUR-ORIGIN-SERVER-WEB-URL>` with the web URL of your Pelican origin server without the `https://` portion. Example: `localhost:8444`
```

* Save the changes to your template dashboard file.

## Import template file to Grafana

> 1. Go to home page of Grafana

Reviewer: ```suggestion
1. Go to the home page of Grafana
```

> 2. Click **Dashboards** in the left-side menu. If there is no menu on the left-side, click the icon of stacked bars to toggle the menu, which is located on the top-left side of the page next to **Home**

Reviewer: ```suggestion
2. Click **Dashboards** on the left-side menu. If there is no menu on the left-side, click the icon of stacked bars to toggle the menu, which is located on the top-left side of the page next to **Home**
```

Author: Grafana documentation uses `in` instead of `on` so I'm keeping up with the same style here

> 3. Click **New** button on the top-right side. Select **Import** option in the dropdown menu.

Reviewer: ```suggestion
3. Click the **New** button on the top-right side. Select the **Import** option in the dropdown menu.
```

4. Drag and drop the template JSON file from your computer to the upload section of the page. It should look like the screenshot below.

    <ExportedImage src={"/pelican/import-dashboard.png"} alt={"Screenshot of Grafana page to import dashboard"} />

5. Click **Import**

Grafana should then redirect you to the page of the import dashboard, which should look like the screenshot below.

<ExportedImage src={"/pelican/origin-dashboard.png"} alt={"Screenshot of Grafana page to import dashboard"} />

> You may notice that some panels have `No data` warning. This should go away once you have run your server for a couple of miniutes and refresh the page. It takes about 15s for Pelican to collect the initial data points and there could be delay in visualizing the data.

Reviewer: ```suggestion
You may notice that some panels have a `No data` warning. This should go away once you have run your server for a couple of minutes and refreshed the page. It takes about 15s for Pelican to collect the initial data points and there could be delay in visualizing the data.
```
